### DISCUSSION TITLE
How to remove unreferenced resources by using C-API read/write memory instead of filesystem

### DISCUSSION THREAD

**Discussion by mslichao** [qpdf/qpdf#1131] [Q&A]
I want to used C# to call C-API to remove unreferenced resources from PDF. The input PDF is already read to byte array in memory, and I want to write the output into another byte array.
I tried `qpdfjob_run_from_json` interface, it works with `removeUnreferencedResources` flag set to `yes`. But this interface needs the inputFile/outputFile be file path.
Then I tried to use the interface in `qpdf-c.h` which support read/write memory, however I didn't find how to set `removeUnreferencedResources` flag to `yes`.
Does anyone know if it is possible to read/write memory, and set `removeUnreferencedResources` flag to `yes` at same time?

**Comment by m-holger** [comment]
qpdf-c.h only exposes a small part of the qpdf library. Unfortunately it does not include the method to remove unreferenced resources. There is a way to extend the C API, but it does involve some C++ programming. 

If you want to go down that road have a look at the files `extend-c-api*.*` in the examples folder. I can give you some pointers on how to do the C++ code if needed.
